h5netcdf/tests/test_h5netcdf.py

```diff
@@ -2404,6 +2404,181 @@ def test_enum_type_creation(tmp_local_or_remote_netcdf, netcdf_write_module):
+    with legacyapi.Dataset(
+        tmp_local_or_remote_netcdf, "r", auto_complex=auto_complex
+    ) as ds:
+        assert ("_PFNC_DOUBLE_COMPLEX_TYPE" in ds.cmptypes) == auto_complex
+        assert array_equal(ds["complex"], complex_array)
+
+    if not tmp_local_or_remote_netcdf.startswith(remote_h5):
+        # netCDF4 only reads variable, if auto_complex == True
+        with netCDF4.Dataset(
+            tmp_local_or_remote_netcdf, "r", auto_complex=auto_complex
+        ) as ds:
+            assert ("_PFNC_DOUBLE_COMPLEX_TYPE" in ds.cmptypes) == auto_complex
+            if auto_complex:
+                assert array_equal(ds["complex"], complex_array)
+            else:
+                assert "complex" not in ds.variables
+
+
+@pytest.mark.skipif(
+    version.parse(netCDF4.__version__) < version.parse("1.7.0"),
+    reason="does not work before netCDF4 v1.7.0",
+)
+def test_nc_complex_compatibility(tmp_local_or_remote_netcdf, netcdf_write_module):
+    complex_array = np.array([0 + 0j, 1 + 0j, 0 + 1j, 1 + 1j, 0.25 + 0.75j])
+    kwargs = {}
+    if (
+        netcdf_write_module.__name__ == "netCDF4"
+        and tmp_local_or_remote_netcdf.startswith(remote_h5)
+    ):
+        pytest.skip("does not work for netCDF4")
+
+    if netcdf_write_module.__name__ == "netCDF4":
+        kwargs.update(auto_complex=True)
+    with netcdf_write_module.Dataset(tmp_local_or_remote_netcdf, "w", **kwargs) as ds:
+        ds.createDimension("x", size=len(complex_array))
+        var = ds.createVariable("data", "c16", ("x",))
+        var[:] = complex_array
+
+    with legacyapi.Dataset(tmp_local_or_remote_netcdf, "r") as ds:
+        dtype = ds.cmptypes["_PFNC_DOUBLE_COMPLEX_TYPE"]
+        assert isinstance(dtype, h5netcdf.legacyapi.CompoundType)
+        assert dtype.name == "_PFNC_DOUBLE_COMPLEX_TYPE"
+        assert array_equal(ds["data"][:], complex_array)
+
+    if not tmp_local_or_remote_netcdf.startswith(remote_h5):
+        with netCDF4.Dataset(tmp_local_or_remote_netcdf, "r", auto_complex=True) as ds:
+            dtype = ds.cmptypes["_PFNC_DOUBLE_COMPLEX_TYPE"]
+            assert isinstance(dtype, netCDF4._netCDF4.CompoundType)
+            assert array_equal(ds["data"][:], complex_array)
+
+
+@pytest.mark.skipif(
+    version.parse(netCDF4.__version__) < version.parse("1.7.0"),
+    reason="does not work before netCDF4 v1.7.0",
+)
+@pytest.mark.parametrize("dtype", ["c4", "c8", "c16", "c32"])
+def test_complex_type_creation_errors(tmp_local_netcdf, netcdf_write_module, dtype):
+    complex_array = np.array([0 + 0j, 1 + 0j, 0 + 1j, 1 + 1j, 0.25 + 0.75j])
+    kwargs = dict(auto_complex=True)
+
+    with netcdf_write_module.Dataset(tmp_local_netcdf, "w", **kwargs) as ds:
+        ds.createDimension("x", size=len(complex_array))
+        if dtype == "c4":
+            try:
+                var = ds.createVariable("data", dtype, ("x",))
+            except TypeError as e:
+                assert e.args[0] == "data type 'c4' not understood"
```

Comment:
pytest.raises is the standard way to write this: https://seaborn.pydata.org/tutorial/color_palettes.html#perceptually-uniform-palettes